processor/resourcedetectionprocessor/internal/resourcedetection.go

```diff
@@ -133,9 +132,10 @@ func resourceToMap(resource pdata.Resource) map[string]interface{} {
 			mp[k] = v.DoubleVal()
 		case pdata.AttributeValueSTRING:
 			mp[k] = v.StringVal()
+		case pdata.AttributeValueMAP:
+			mp[k] = AttributesToMap(v.MapVal())
```

Comment:
Thanks for adding. Not for this PR, but just a reminder that we should move this function into `pdata` or `pdatautil` at some point in the future.